activerecord/lib/active_record/query_cache.rb

```diff
@@ -26,19 +26,15 @@ def uncached(&block)
     end
 
     def self.run
-      ActiveRecord::Base.connection_handler.connection_pool_list.map do |pool|
-        caching_was_enabled = pool.query_cache_enabled
-
-        pool.enable_query_cache!
-
-        [pool, caching_was_enabled]
-      end
+      ActiveRecord::Base.connection_handler.connection_pool_list.select do |pool|
+        unless pool.query_cache_enabled
+           pool.enable_query_cache!
```

Comment:
`ConnectionPool#enable_query_cache!` returns nil when there's no active connection:
https://github.com/rails/rails/blob/f339b098b20c7f6675d375cfa29828a911fe1b53/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L36

The query cache will be enabled but the pool won't be returned, so the complete hook won't disable it... which will cause a regression of https://github.com/rails/rails/issues/26666 🐛

Reply:
It's probably clearer to find the connection pools and enable the query cache on them as separate steps:

```ruby
ActiveRecord::Base.connection_handler.connection_pool_list
  .reject(&:query_cache_enabled).each(&:enable_query_cache!)
```

Could you update [this test](https://github.com/rails/rails/blob/f339b098b20c7f6675d375cfa29828a911fe1b53/activerecord/test/cases/query_cache_test.rb#L438) to assert that the query cache is disabled afterwards too?